src/app/pages/common/ix-forms/services/ix-validators.service.ts

```diff
@@ -0,0 +1,204 @@
+      if (errors?.email) {
+        errors.email = errMessage;
+      }
+      return errors;
+    };
+  }
+
+  /**
+   * This method applies the Validators.minLength validator to the input
+   * but applies errMessage as the custom error message.
+   * @param minLength The minium length allowed
+   * @param errMessage The error message if this validator fails
+   * @returns A validator function that applies Validators.minLength and if failed,
+   * applies errMessage as the custom error message.
+   */
+  minLength(minLength: number, errMessage?: string): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const errors = Validators.minLength(minLength)(control);
+      if (errors?.minlength) {
+        errors.minlength.message = errMessage;
+      }
+      return errors;
+    };
+  }
+
+  /**
+   * This method applies the Validators.maxLength validator to the input
+   * but applies errMessage as the custom error message.
+   * @param maxLength The maximum length allowed
+   * @param errMessage The error message if this validator fails
+   * @returns A validator function that applies Validators.maxLength and if failed,
+   * applies errMessage as the custom error message.
+   */
+  maxLength(maxLength: number, errMessage?: string): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const errors = Validators.maxLength(maxLength)(control);
+      if (errors?.maxlength) {
+        errors.maxlength.message = errMessage;
+      }
+      return errors;
+    };
+  }
+
+  /**
+   * This method applies the Validators.pattern validator to the input
+   * but applies errMessage as the custom error message.
+   * @param pattern The pattern to check the control value against
+   * @param errMessage The error message if this validator fails
+   * @returns A validator function that applies Validators.pattern and if failed,
+   * applies errMessage as the custom error message.
+   */
+  pattern(pattern: string | RegExp, errMessage?: string): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      const errors = Validators.pattern(pattern)(control);
+      if (errors?.pattern) {
+        errors.pattern.message = errMessage;
+      }
+      return errors;
+    };
+  }
+
+  /**
+   * For inclusion or Validators.nullValidator
+   * @returns Validators.nullValidator
+   */
+  nullValidator(): ValidatorFn {
+    return Validators.nullValidator;
+  }
+
+  /**
+   * For inclusion or Validators.compose
+   * @returns the result of Validators.compose called on the passed validator functions array
+   */
+  compose(validators: ValidatorFn[]): ValidatorFn | null {
+    return Validators.compose(validators);
+  }
+
+  /**
+   * For inclusion or Validators.composeAsync
+   * @returns the result of Validators.composeAsync called on the passed validator functions array
+   */
+  composeAsync(validators: AsyncValidatorFn[]): AsyncValidatorFn | null {
```

Comment:
`composeAsync` and `compose` are not really needed, could be used directly from Angular's `Validators`.